create_test.go

```diff
@@ -0,0 +1,98 @@
+package statusresource
+
+import (
+	"testing"
+	"time"
+
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/google/go-cmp/cmp"
+)
+
+func Test_removeTimesFromNodes(t *testing.T) {
+	testCases := []struct {
+		name          string
+		nodes         []providerv1alpha1.StatusClusterNode
+		expectedNodes []providerv1alpha1.StatusClusterNode
+	}{
+		{
+			name: "case 0: Ensure that name, version and labels are kept when dropping last heartbeat and last transitioning time",
+			nodes: []providerv1alpha1.StatusClusterNode{
+				{
+					Labels: map[string]string{
+						"foo": "bar",
+						"bar": "baz",
+						"baz": "foo",
+					},
+
+					LastHeartbeatTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Unix(123456, 0),
+					},
+					LastTransitionTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Unix(123000, 0),
+					},
+					Name:    "worker-1234",
+					Version: "1.2.3",
+				},
+				{
+					Labels: map[string]string{
+						"abc": "123",
+						"def": "456",
+						"ghi": "789",
+					},
+
+					LastHeartbeatTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Unix(234567, 0),
+					},
+					LastTransitionTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Unix(234500, 0),
+					},
+					Name:    "worker-2345",
+					Version: "2.3.4",
+				},
+			},
+			expectedNodes: []providerv1alpha1.StatusClusterNode{
+				{
+					Labels: map[string]string{
+						"foo": "bar",
+						"bar": "baz",
+						"baz": "foo",
+					},
+
+					LastHeartbeatTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Time{},
+					},
+					LastTransitionTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Time{},
+					},
+					Name:    "worker-1234",
+					Version: "1.2.3",
+				},
+				{
+					Labels: map[string]string{
+						"abc": "123",
+						"def": "456",
+						"ghi": "789",
+					},
+
+					LastHeartbeatTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Time{},
+					},
+					LastTransitionTime: providerv1alpha1.DeepCopyTime{
+						Time: time.Time{},
+					},
+					Name:    "worker-2345",
+					Version: "2.3.4",
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			nodes := removeTimesFromNodes(tc.nodes)
+			if diff := cmp.Diff(nodes, tc.expectedNodes); diff != "" {
```

Comment:
Personal note, I really hate mixing functionality and conditionals in one line. I prefer to keep that separated and not mix it. Working with it and reading it is way easier down the road IMO.